great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -224,18 +224,19 @@ def build_batch_spec(
         batch_spec: BatchSpec = super().build_batch_spec(
             batch_definition=batch_definition
         )
-        if runtime_parameters.get("batch_data") is not None:
+        if "batch_data" in runtime_parameters:
             batch_spec["batch_data"] = runtime_parameters.get("batch_data")
             return RuntimeDataBatchSpec(batch_spec)
-        elif runtime_parameters.get("query"):
+        elif "query" in runtime_parameters:
             batch_spec["query"] = runtime_parameters.get("query")
             return RuntimeQueryBatchSpec(batch_spec)
-        elif runtime_parameters.get("path"):
-            path = runtime_parameters.get("path")
+        elif "path" in runtime_parameters:
+            path: str = runtime_parameters["path"]
             batch_spec["path"] = path
-            parsed_url = urlparse(path)
-            if "s3" in parsed_url.scheme:
+            if "s3" in path:
                 return S3BatchSpec(batch_spec)
+            elif "blob.core.windows.net" in path:
+                return AzureBatchSpec(batch_spec)
             else:
```

Comment:
This is a general refactor/cleanup that simplifies the logic used to determine which `BatchSpec` to use. Note that I've added `AzureBatchSpec` and removed usage of `urlparse` since a simple `x in y` check fulfills the same behavior without needing an import.

Reply:
😁